sds/src/scanner/mod.rs

```diff
@@ -1130,6 +1131,30 @@ mod test {
         assert_eq!(content, "[GITHUB] ghp_M7H4jxUDDWHP4kZ6A4dxlQYsQIWJuq11T4V5");
     }
 
+    #[test]
+    fn test_nhs_checksum() {
+        let pattern = ".+";
+        let rule = RegexRuleConfig::builder(pattern.to_string())
+            .match_action(MatchAction::Redact {
+                replacement: "[NHS]".to_string(),
+            })
+            .build();
+
+        let rule_with_checksum = RuleConfigBuilder::from(&rule)
+            .validator(NhsCheckDigit)
+            .build();
+
+        let mut content = "1234567881".to_string();
+        // Test matching NHS number with checksum
+        let scanner = ScannerBuilder::new(&[rule_with_checksum.clone()])
+            .with_keywords_should_match_event_paths(true)
```

Comment:
This is not needed, I suggest to not include it so whenever we will remove this option, we will have less cleanup
```suggestion
```